src/tools/ScenarioMeasurement/Startup.Tests/StartupTests.cs

```diff
@@ -43,6 +46,72 @@ public void TestProfileIteration()
+            var pids = new List<int>() { 18627, 18674, 18721, 18768, 18813 };
+            IEnumerable<Counter> counters = parser.Parse(ctfFile, "dotnet", pids, "\"dotnet\" build");
+            int count = 0;
+            foreach (var counter in counters)
+            {
+                Assert.True(counter.Results.Count == pids.Count, $"Counter {counter.Name} is expected to have {pids.Count} results.");
+                count++;
+            }
+            Assert.True(count == 1, "Only Process Time counter should be present.");
+        }
+
+        [Fact]
+        public void TestProcessTimeParserWindows()
+        {
+            string etlFile = Path.Combine(testAssetDirectory, "test-process-time_startup.etl");
+            var parser = new ProcessTimeParser();
+            var pids = new List<int>() { 32752, 6352, 16876, 10500, 17784 };
+            IEnumerable<Counter> counters = parser.Parse(etlFile, "dotnet", pids, "\"dotnet\" build");
+            int count = 0;
+            foreach (var counter in counters)
+            {
+                Assert.True(counter.Results.Count == pids.Count, $"Counter {counter.Name} is expected to have {pids.Count} results.");
+                count++;
+            }
+            Assert.True(count==2, "Both Process Time and Time To Main counter should be present.");
+        }
+
+        [Fact]
+        public void TestTimeToMainParserLinux()
+        {
+            string ctfFile = Path.Combine(testAssetDirectory, "test-time-to-main_startup.trace.zip");
+            var parser = new TimeToMainParser();
+            var pids = new List<int>() { 24352, 24362, 24371, 24380, 24389 };
+            IEnumerable<Counter> counters = parser.Parse(ctfFile, "emptycsconsoletemplate", pids, "\"pub\\emptycsconsoletemplate.exe\"");
+            int count = 0;
+            foreach (var counter in counters)
+            {
+                Assert.True(counter.Results.Count == pids.Count, $"Counter {counter.Name} is expected to have {pids.Count} results.");
+                count++;
+            }
+            Assert.True(count == 2, "Only Time To Main counter should be present.");
+        }
+
+
+        [Fact]
+        public void TestTimeToMainParserWindows()
+        {
+            string etlFile = Path.Combine(testAssetDirectory, "test-time-to-main_startup.etl");
+            var parser = new TimeToMainParser();
+            var pids = new List<int>() { 17036, 21640, 12912, 19764, 11624 };
+            IEnumerable<Counter> counters = parser.Parse(etlFile, "emptycsconsoletemplate", pids, "\"pub\\emptycsconsoletemplate.exe\"");
+            int count = 0;
+            foreach (var counter in counters)
+            {
+                Assert.True(counter.Results.Count == pids.Count, $"Counter {counter.Name} is expected to have {pids.Count} results.");
+                count++;
+            }
+            Assert.True(count == 2, "Both Time To Main and Time To Thread counter should be present.");
+        }
+
+
```

Comment:
This also seems from the name like it should only be testing the time to main parsing, but again we check for two counters.

Reply:
Uh I think it should check 2 counters here on Windows: "time to main" and "time on thread".  (shocked by how many mistakes I made here)

Reply:
You are good here. I mis-read the test names and didn't realize it was testing different parsers. I thought it was supposed to be testing different counters we could get out of the parsers.